### ISSUE TITLE
Refactoring needed

### ISSUE THREAD

**Issue by philipperemy** [Cartucho/mAP#109]
According the flake8, the main function complexity is a whopping 91!

Also, it would be nice to be able to install it with pip and have one function to call with a list of bounding boxes from ground truth and predictions and get a number.

Thanks!

**Comment by Cartucho** [comment]
Yes, you are right?
Do you think you could help me in any of those tasks?

**Comment by HarrisDePerceptron** [comment]
this code needs a redo. looks like written by a 2 year old (SE years)

**Comment by Cartucho** [comment]
Yeah, and it is indeed the case. It should be using classes; it needs documentation and tests...
Currently, I do not have time to do that, but feel free to take the initiative.

**Comment by philipperemy** [comment]
Same I wish I had time.
